plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -240,16 +247,30 @@ public ClickHouseClient(
         this.uuidType = typeManager.getType(new TypeSignature(StandardTypes.UUID));
         this.ipAddressType = typeManager.getType(new TypeSignature(StandardTypes.IPADDRESS));
         JdbcTypeHandle bigintTypeHandle = new JdbcTypeHandle(Types.BIGINT, Optional.of("bigint"), Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty());
-        this.connectorExpressionRewriter = JdbcConnectorExpressionRewriterBuilder.newBuilder()
+        JdbcConnectorExpressionRewriterBuilder connectorExpressionRewriterForJoinBuilder = JdbcConnectorExpressionRewriterBuilder.newBuilder()
                 .addStandardRules(this::quoted)
+                .map("$equal(left, right)").to("left = right")
+                .add(new RewriteIn())
+                .map("$like(value: varchar, pattern: varchar): boolean").to("value LIKE pattern")
+                .map("$not($is_null(value))").to("value IS NOT NULL")
+                .map("$not(value: boolean)").to("NOT value")
+                .map("$is_null(value)").to("value IS NULL")
+                .map("$nullif(first, second)").to("NULLIF(first, second)");
+        this.connectorExpressionRewriterForJoin = connectorExpressionRewriterForJoinBuilder.build();
+        this.connectorExpressionRewriter = JdbcConnectorExpressionRewriterBuilder.newBuilder(connectorExpressionRewriterForJoinBuilder)
+                .map("$not_equal(left, right)").to("left <> right")
+                .map("$less_than(left, right)").to("left < right")
+                .map("$less_than_or_equal(left, right)").to("left <= right")
+                .map("$greater_than(left, right)").to("left > right")
+                .map("$greater_than_or_equal(left, right)").to("left >= right")
```

Comment:
These would be applied for all the data types incl double and real as well - What would be its behaviour wrt to NaN and Infinity ?

Reply:
Relates to https://github.com/trinodb/trino/pull/23496 
pushdowns for `real` are disabled,
inequalities for `double` to be tested